### PR TITLE
Andy's part 4 submission

### PR DESCRIPTION
**Pull request by amitchell94**

### CODE REVIEW

File: src/part_04/1_questions [hunk 1]
```diff
@@ -2,48 +2,116 @@ Answer the following questions:
...
+ An a array is a collection of variables of the same type which are contained in one object.
+
 2) What the code required to create (instantiate) a String array with size 10 named "cats"?
 
+ String[] cats = new String[10];
+
 3) What is a multidimensional array?
 
+ A multidimensional array can be thought of as an array that contains one or more arrays.
+
```
Comment (T-Davis) [inline src/part_04/1_questions]: I'd say two or more arrays

File: src/part_04/1_questions [hunk 2]
```diff
@@ -2,48 +2,116 @@ Answer the following questions:
...
 
+ String dog10 = dogs[9];
+
 6) What is an ArrayIndexOutOfBoundsException?
 
+ When you are trying to reference a variable at an index which is outside the range of the array.
+
 7) Using one line of code, demonstrate how to retrieve the second to the last item from a int array.
 
+ int secondToLastInt = intArray[intArray-1];
```
Comment (T-Davis) [inline src/part_04/1_questions]: Close but not quite

File: src/part_04/1_questions [hunk 3]
```diff
@@ -2,48 +2,116 @@ Answer the following questions:
...
+ for(int i = 0; i < array.length; i++) {
+    System.out.println(array[i]);
+ }
+
 13) Demonstrate how to iterate through an array, printing each value, using a while loop.
 
+ int currentIndex = 0;
+
+ while (currentIndex < array.length) {
+    System.out.println(array[i]);
```
Comment (T-Davis) [inline src/part_04/1_questions]: What is i?

File: src/part_04/1_questions (hunk 1 above)
Comment (amitchell94) [reply src/part_04/1_questions]: Yep, good point. Thanks

File: src/part_04/1_questions (hunk 2 above)
Comment (amitchell94) [reply src/part_04/1_questions]: Should've had .length-2 in there right?

File: src/part_04/Exercise_06.java [hunk 4]
```diff
@@ -10,4 +14,26 @@
...
+        for (int i = 0; i < 101; i++) {
+            queue.add(i);
+        }
+
+        boolean doPrint = true;
+        while(!queue.isEmpty()) {
+            if (doPrint) {
+                System.out.println(queue.poll());
+            } else {
+                ((ArrayDeque<Integer>) queue).removeFirst();
```
Comment (T-Davis) [inline src/part_04/Exercise_06.java]: queue.remove() would have been simpler

File: src/part_04/1_questions (hunk 3 above)
Comment (amitchell94) [reply src/part_04/1_questions]: i should be currentIndex. Whoops!

File: src/part_04/Exercise_06.java (hunk 4 above)
Comment (amitchell94) [reply src/part_04/Exercise_06.java]: Good point, not sure why I didn't use that

**Pull request merged**